### PR TITLE
Mostly safe refactorings of CloudDebugHistoricalSnapshots

### PR DESCRIPTION
**Pull request by elharo**
@patflynn This cl performs a number of relatively safe, automatic refactorings in the CloudDebugHistoricalSnapshots class and a few related classes. It also expands on the intenral documentation in the comments. The main refactoring that was performed was pulling out a lot of anonymous inner classes into non-anonymous inner classes. 

Step 2 is going to be to further separate the model from the GUI so we can better test this code.

### CODE REVIEW

**Comment by elharo** [comment]
@joaoandremartins this should probably go in before work on several outstanding bugs since it cleans up a lot of the code where the bugs seemj to be hiding.

**Comment by patflynn** [comment]
I'\m looking at this next. Should have comments this afternoon.

File: core-plugin/src/com/google/gct/idea/debugger/CloudBreakpointHandler.java [hunk 1]
```diff
@@ -79,8 +79,9 @@ public CloudBreakpointHandler(@NotNull CloudDebugProcess process) {
    * Called when the user "reactivates" an existing snapshot to create a new breakpoint, this clones state into a new
    * breakpoint. It is different from "createIdeRepresentationsIfNecessary" in the following respects:
    * <p/>
-   * 1. it only operates on final state breakpoints. 2. it always clones them into a new IDE breakpoint 3. it does not
-   * set "created by server = true", so when control flow comes back into register, it will register with the server.
+   * 1. it only operates on final state breakpoints.
+   * 2. it always clones them into a new IDE breakpoint
```
Comment (patflynn) [inline core-plugin/src/com/google/gct/idea/debugger/CloudBreakpointHandler.java]: Should these bullet points be capitalized?  Also 2) is missing a period at the end.

File: core-plugin/src/com/google/gct/idea/debugger/ui/CloudDebugHistoricalSnapshots.java [hunk 2]
```diff
@@ -106,34 +109,11 @@
...
-          JComponent jc = (JComponent)c;
-          Breakpoint breakpoint = CloudDebugHistoricalSnapshots.this.getModel().getBreakpoints().get(row);
-          jc.setToolTipText(BreakpointUtil.getUserErrorMessage(breakpoint.getStatus()));
-        }
-        return c;
-      }
-    };
+    myTable = new CloudDebuggerTable();
 
+    // todo I think all of this can be pushed into CloudDebuggerTable unless there's some weird
```
Comment (patflynn) [inline core-plugin/src/com/google/gct/idea/debugger/ui/CloudDebugHistoricalSnapshots.java]: let's put a colon after todos. 'todo:..'

File: core-plugin/src/com/google/gct/idea/debugger/ui/CloudDebugHistoricalSnapshots.java [hunk 3]
```diff
@@ -242,6 +163,8 @@ public void beforeSessionResume() {
 
   @Override
   public void dispose() {
+    // todo can we clear pending snapshots here to keep them from getting duped on reconnect?
```
Comment (patflynn) [inline core-plugin/src/com/google/gct/idea/debugger/ui/CloudDebugHistoricalSnapshots.java]: here.

File: core-plugin/src/com/google/gct/idea/debugger/CloudBreakpointHandler.java (hunk 1 above)
Comment (elharo) [inline core-plugin/src/com/google/gct/idea/debugger/CloudBreakpointHandler.java]: done

**Comment by patflynn** [comment]
Huge improvement. Thanks.

LGTM

File: core-plugin/src/com/google/gct/idea/debugger/ui/CloudDebugHistoricalSnapshots.java (hunk 2 above)
Comment (elharo) [inline core-plugin/src/com/google/gct/idea/debugger/ui/CloudDebugHistoricalSnapshots.java]: done

**Pull request merged**